### PR TITLE
improve ci

### LINKED ISSUE #675: Improve CI
Travis config should avoid running the CI outside PRs and main branches and it should take into account keywords like `[skip-ci]` and `[wip]`

### PR DESCRIPTION
**Pull request by berthieresteban**
## What does this PR do ?
fix #675

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -36,6 +36,10 @@ jobs:
     # ---------------------------------------
     - stage: lint
       name: 'Linter'
+      if: (type IN (pull_request))
+        AND (branch IN (master, 4-dev, 3-dev, 2-dev, 2-stable))
+        AND commit_message !~ /(skip-ci|wip)/
```
Comment (Yoann-Abbes) [inline .travis.yml:41]: Don't you should write something like
```
 name: 'Linter'
    if: "!contains(github.event.head_commit.message, 'skip-ci')"```
```
instead? @alexandrebouthinon 
https://github.com/marketplace/actions/skip-based-on-commit-message

**Review by Yoann-Abbes — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (berthieresteban) [reply .travis.yml:41]: I used travis [conditionnal builds](https://docs.travis-ci.com/user/conditions-v1)

Comment (Yoann-Abbes) [reply .travis.yml:41]: You r right. I dont know why I thought it was a travis configuration file -_-

**Review by Yoann-Abbes — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (alexandrebouthinon) [reply .travis.yml:41]: No need to add an explicit condition, Travis handle it for you https://docs.travis-ci.com/user/customizing-the-build/#skipping-a-build

Comment (alexandrebouthinon) [reply .travis.yml:41]: There is a better way to do that let's talk about it on Discord

**Review by alexandrebouthinon — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -36,6 +40,10 @@ jobs:
     # ---------------------------------------
     - stage: lint
       name: 'Linter'
+      if: (type IN (pull_request))
+        AND (branch IN (master, 4-dev, 3-dev, 2-dev, 2-stable))
```
Comment (xbill82) [inline .travis.yml:44]: Aren't these 2 lines redundant with line 29? By the way, the `type` condition is not the same: `pull_request` here, VS `cron|pull_request`.

I would rather delete these 2 and just keep the `skip-ci|wip` condition.

**Review by xbill82 — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (alexandrebouthinon) [reply .travis.yml:44]: Delete the whole condition IMO
@xbill82 about skipping CI build in Travis https://github.com/kuzzleio/kuzzle-admin-console/pull/854#discussion_r535090753

**Review by alexandrebouthinon — CHANGES_REQUESTED** [review]
You can remove the condition on lint job since we set it for the lint stage

**Comment by alexandrebouthinon** [comment]
No more needed since we'll migrate to Github Actions very soon

**Pull request closed without merging**